roles/sahara/tasks/main.yml

```diff
@@ -0,0 +1,81 @@
+---
+- name: create sahara user
+  user: name=sahara comment=sahara shell=/bin/false system=yes home=/nonexistent
+        createhome=no
+
+- name: create sahara config directories
+  file: dest={{ item }} state=directory owner=root group=sahara mode=0750
+  with_items:
+    - /etc/sahara
+
+- name: create sahara log directory
+  file: dest=/var/log/sahara state=directory mode=0755 owner=sahara
+
+- name: create sahara temp directory
+  file: dest=/opt/stack/sahara state=directory mode=0755 owner=sahara
+
+- debug: msg="path of sahara virtual env {{ sahara.virtualenv }}"
+
+- name: create virtualenv for sahara
+  command: "virtualenv {{ sahara.virtualenv }}"
+
+- name: fetch sahara-juno stable tarball
+  get_url: url={{ sahara.juno_url }} dest=/opt/stack/sahara/sahara-stable-juno.tar.gz
+
+- name: install sahara-juno 
+  command: "{{ sahara.virtualenv }}/bin/pip install /opt/stack/sahara/sahara-stable-juno.tar.gz"
+
+- name: install sahara-juno mysql module 
+  command: "{{ sahara.virtualenv }}/bin/pip install mysql-python"
```

Comment:
Correct if I am wrong. If defaults.yml "openstack_install_method: 'source' " then openstack-source is used and if its "package" then openstack-package role is used right?

If I will use the openstack-source role then it should be set to source, otherwise you guys will need to package and upload to your repo in order to use package method. 

Let me know and I'll start work accordingly.